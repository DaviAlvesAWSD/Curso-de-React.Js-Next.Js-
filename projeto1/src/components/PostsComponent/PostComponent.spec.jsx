import { render, screen } from '@testing-library/react';

import { PostsComponent } from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'title 1',
      body: 'body 1',
      cover: 'img/img1.png',
    },
    {
      id: 2,
      title: 'title 2',
      body: 'body 2',
      cover: 'img/img2.png',
    },
    {
      id: 3,
      title: 'title 3',
      body: 'body 3',
      cover: 'img/img3.png',
    },
  ],
};

describe('<PostComponent />', () => {
  it('Should render posts', () => {
    render(<PostsComponent {...props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: 'title 3' })).toHaveAttribute('src', 'img/img3.png');
  });

  it('Should not render posts', () => {
    render(<PostsComponent />);

    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
    expect(screen.queryByRole('img', { name: /title/i })).not.toBeInTheDocument();
    expect(screen.queryByText(/body/i)).not.toBeInTheDocument();
    expect(screen.queryByRole('img', { name: /title/i })).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostsComponent {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
