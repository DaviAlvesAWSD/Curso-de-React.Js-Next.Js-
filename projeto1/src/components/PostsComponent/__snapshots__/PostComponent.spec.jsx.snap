// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostComponent /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 2
      </h2>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title 3
      </h2>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
